sklearn/tests/test_naive_bayes.py

```diff
@@ -945,3 +952,678 @@ def test_n_features_deprecation(Estimator):
+        nb_estimators=[
+            ("b1", BernoulliNB(binarize=2), [1]),
+            ("m1", MultinomialNB(), [0, 2, 3]),
+        ]
+    )
+    clf1.fit(X2, y2)
+    for se in clf1.named_estimators_:
+        assert_array_almost_equal(
+            clf1.class_count_, clf1.named_estimators_[se].class_count_, 8
+        )
+        assert_array_almost_equal(clf1.classes_, clf1.named_estimators_[se].classes_, 8)
+        assert_array_almost_equal(
+            np.log(clf1.class_prior_), clf1.named_estimators_[se].class_log_prior_, 8
+        )
+
+
+def test_cwnb_params():
+    # Can get and set subestimators' parameters through name__paramname
+    # clone() works on ColumnwiseNB
+    clf1 = ColumnwiseNB(
+        nb_estimators=[
+            ("b1", BernoulliNB(alpha=0.2, binarize=2), [1]),
+            ("m1", MultinomialNB(class_prior=[0.2, 0.2, 0.6]), [0, 2, 3]),
+        ]
+    )
+    clf1.fit(X2, y2)
+    p = clf1.get_params(deep=True)
+    assert p["b1__alpha"] == 0.2
+    assert p["b1__binarize"] == 2
+    assert p["m1__class_prior"] == [0.2, 0.2, 0.6]
+    clf1.set_params(b1__alpha=123, m1__class_prior=[0.3, 0.3, 0.4])
+    assert clf1.nb_estimators[0][1].alpha == 123
+    assert_array_equal(clf1.nb_estimators[1][1].class_prior, [0.3, 0.3, 0.4])
+    # After cloning and fitting, we can check through named_estimators, which
+    # maps to fitted estimators_:
+    clf2 = clone(clf1).fit(X2, y2)
+    assert clf2.named_estimators_["b1"].alpha == 123
+    assert_array_equal(clf2.named_estimators_["m1"].class_prior, [0.3, 0.3, 0.4])
+    assert id(clf2.named_estimators_["b1"]) != id(clf1.named_estimators_["b1"])
+
+
+def test_cwnb_n_jobs():
+    # n_jobs: same results wether with it or without
+    clf1 = ColumnwiseNB(
+        nb_estimators=[
+            ("b1", BernoulliNB(binarize=2), [1]),
+            ("b2", BernoulliNB(binarize=2), [1]),
+            ("m1", MultinomialNB(), [0, 2, 3]),
+            ("m3", MultinomialNB(), slice(10, None)),
+        ],
+        n_jobs=4,
+    )
+    clf2 = ColumnwiseNB(
+        nb_estimators=[
+            ("b1", BernoulliNB(binarize=2), [1]),
+            ("b2", BernoulliNB(binarize=2), [1]),
+            ("m1", MultinomialNB(), [0, 2, 3]),
+            ("m3", MultinomialNB(), slice(10, None)),
+        ]
+    )
+    clf1.partial_fit(X2, y2, classes=np.unique(y2))
+    clf2.partial_fit(X2, y2, classes=np.unique(y2))
+
+    assert_array_almost_equal(
+        clf1._joint_log_likelihood(X2), clf2._joint_log_likelihood(X2), 8
+    )
+    assert_array_almost_equal(clf1.predict_log_proba(X2), clf2.predict_log_proba(X2), 8)
+    assert_array_equal(clf1.predict(X2), clf2.predict(X2))
+
+
+def test_cwnb_example():
+    # Test the Example from ColumnwiseNB docstring in naive_bayes.py
+    import numpy as np
+
+    rng = np.random.RandomState(1)
+    X = rng.randint(5, size=(6, 100))
+    y = np.array([0, 0, 1, 1, 2, 2])
+    from sklearn.naive_bayes import MultinomialNB, GaussianNB, ColumnwiseNB
```

Comment:
import at the top

Reply:
The line is removed. It was there because I copied-and-pasted the example from the docstring.